lib/expression.js

```diff
@@ -345,7 +352,7 @@ CronExpression._parseField = function _parseField (field, value, constraints) {
       return stack;
     }
 
-    return +val;
+    return field === 'dayOfMonth' && val === 'L'? val : +val;
```

Comment:
I need to test this myself, but we could possibly generalise in this way:

```suggestion
    return isNan(+val) ? val : +val;
```

If this doesn't work, ignore my suggestion.

Reply:
isNan is generating a error . Saying isNan is not defined. Shouldnt it be **isNaN**